lib/src/util/bolt_helpers.cc

```diff
@@ -0,0 +1,188 @@
+#include <pxp-agent/util/bolt_helpers.hpp>
+#include <pxp-agent/configuration.hpp>
+#include <pxp-agent/module.hpp>
+
+#include <boost/algorithm/hex.hpp>
+
+#define LEATHERMAN_LOGGING_NAMESPACE "puppetlabs.pxp_agent.util.bolt_helpers"
+#include <leatherman/logging/logging.hpp>
+
+#include <openssl/evp.h>
+
+namespace fs       = boost::filesystem;
+namespace alg      = boost::algorithm;
+namespace lth_curl = leatherman::curl;
+namespace lth_jc   = leatherman::json_container;
+namespace lth_loc  = leatherman::locale;
+
+namespace PXPAgent {
+namespace Util {
+
+
+    // NIX_DIR_PERMS is defined in pxp-agent/configuration
+    #define NIX_TASK_FILE_PERMS NIX_DIR_PERMS
+
+    // Downloads a file if it does not already exist on the filesystem. A check is made
+    // on the filesystem to determine if the file at destination already exists and if
+    // it already matches the sha256 provided with the file. If the file already exists
+    // the function immediately returns.
+    //
+    // If the file does not exist attempt to download with leatherman.curl. Once the
+    // download finishes a sha256 check occurs to ensure file contents are correct. Then
+    // the file is moved to destination with boost::filesystem::rename.
+    //
+    // If the destination param is empty: the function assumes the file should
```

Comment:
I don't see logic to support this last comment block. Didn't you move this elsewhere, like into the caller, per the other comments?

Reply:
oh, yeah, whoops